src/main/java/by/gsu/epamlab/AbstractPurchase.java

```diff
@@ -0,0 +1,48 @@
+package by.gsu.epamlab;
+
+public abstract class AbstractPurchase implements Comparable<AbstractPurchase> {
+    private final Product product;
+    private int numberUnits;
+
+    public AbstractPurchase() {
+        product = new Product();
+    }
+
+    public AbstractPurchase(Product product, int numberUnits) {
+        this.product = product;
+        this.numberUnits = numberUnits;
+    }
+
+    public Product getProduct() {
+        return product;
+    }
+
+    public int getNumberUnits() {
+        return numberUnits;
+    }
+
+    public void setNumberUnits(int numberUnits) {
+        this.numberUnits = numberUnits;
+    }
+
+    abstract Byn abstractCost(Byn baseCost);
```

Comment:
```suggestion
    protected abstract Byn abstractCost(Byn baseCost);
```
имя метода абсурдное (